crates/matrix-sdk/src/authentication/service.rs

```diff
@@ -811,20 +842,148 @@ pub(crate) mod tests {
         assert!(matches!(error, AuthenticationError::Generic { .. }));
     }
 
+    #[async_test]
+    #[cfg(feature = "experimental-oidc")]
```

Comment:
Honestly, I'm not 100% sure if these new tests will run or not on CI given the feature flag? I've been running all of the tests in this file using which is why I'm asking.

```
cargo test -p matrix-sdk --features experimental-sliding-sync,experimental-oidc authentication::service
```

Reply:
The "all crates" CI steps will run all the tests with the necessary features as required by the combination of crates in the workspace; because we have the bindings and the UI crate, that means `experimental-sliding-sync` and `experimental-oidc` are enabled by default. And you can check that by seeing that some of the oidc tests are run in those CI steps. So this feature guard should work.

You can also double-check if the following command line runs the test:

```
cargo nextest run --workspace --exclude matrix-sdk-integration-testing --features testing -- test_name
```